### PR TITLE
Added logGraph for graph visualisation in TensorBoard

### PR DESCRIPTION
**Pull request by jcberentsen**

### CODE REVIEW

**Comment by blackgnezdo** [comment]
Jenkins test this please.

File: tensorflow-logging/src/TensorFlow/Logging.hs [hunk 1]
```diff
@@ -133,6 +134,14 @@ logSummary writer step' summaryProto = do
                          & summary .~ summaryProto
                     )
 
+-- | Logs the graph for the given (Build a)
```
Comment (blackgnezdo) [inline tensorflow-logging/src/TensorFlow/Logging.hs]: I think @Build a@ would look nicer, also, please add a trailing ".".

File: tensorflow-logging/src/TensorFlow/Logging.hs [hunk 2]
```diff
@@ -133,6 +134,14 @@ logSummary writer step' summaryProto = do
...
+logGraph :: MonadIO m => EventWriter -> Build a -> m ()
+logGraph writer build = do
+  let graph = asGraphDef build
+  let graphBytes = encodeMessage graph
```
Comment (blackgnezdo) [inline tensorflow-logging/src/TensorFlow/Logging.hs]: The let here and below not needed, just keep everything in one let block.

File: tensorflow-logging/src/TensorFlow/Logging.hs [hunk 3]
```diff
@@ -64,10 +65,10 @@ import Data.Time.Clock.POSIX (utcTimeToPOSIXSeconds)
 import Lens.Family2 ((.~), (&))
 import Network.HostName (getHostName)
 import Proto.Tensorflow.Core.Framework.Summary (Summary)
-import Proto.Tensorflow.Core.Util.Event (Event, fileVersion, step, summary, wallTime)
+import Proto.Tensorflow.Core.Util.Event (Event, fileVersion, step, summary, wallTime, graphDef)
```
Comment (blackgnezdo) [inline tensorflow-logging/src/TensorFlow/Logging.hs]: Could you move graphDef closer to the front? We try to keep things sorted. Also in the import below.

Comment (jcberentsen) [inline tensorflow-logging/src/TensorFlow/Logging.hs]: Added test and tidied up.

File: tensorflow-logging/src/TensorFlow/Logging.hs (hunk 2 above)
Comment (jcberentsen) [inline tensorflow-logging/src/TensorFlow/Logging.hs]: Fixed

**Comment by blackgnezdo** [comment]
Jenkins test this please.

**Comment by fkm3** [comment]
Jenkins, test this please.

**Comment by blackgnezdo** [comment]
Jenkins, test this please.

**Comment by jcberentsen** [comment]
Test has been added

**Comment by judah** [comment]
Jenkins, test this please

**Comment by blackgnezdo** [comment]
Thanks for the PR!

**Pull request merged**